Add Warehouse FAQ (#1003)
rudderlabs/rudderstack-docs: docs/destinations/warehouse-destinations/faq.mdx

- For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
- For **other** destinations, the table name is truncated after 127 characters.

### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?

> RudderStack automatically handles the data type conversion for some common scenarios **depending on the warehouse column data type**.

Author: ```suggestion
RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
```

> | Incoming data type | Warehouse column data type | Notes |

Author: ```suggestion
| Scenario | Incoming data type | Warehouse column data type | Notes |
```

> | :----| :------| :------|

Author: ```suggestion
| :-----| :----| :------| :------|
```

> | Integer <br /> Float <br /> JSON | String | RudderStack automatically stringifies the non-string data. |

Author: ```suggestion
| All data types **except** String, Text, or JSON | String | RudderStack automatically stringifies the incoming data. |
```

> | All data types **except** String, Text, or JSON | String | RudderStack automatically stringifies the incoming data. |

Reviewer: ```suggestion
| All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
```

> | All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |

Author: ```suggestion
| 1 | All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
```

> | Integer | Float | - | 

Author: ```suggestion
| Integer <br /> BigInt | Float | - | 
```

> | Float | Integer | - |

Author: ```suggestion
| Float | Integer <br /> BigInt | - |
```

> | Integer <br /> BigInt | Float | - | 

Author: ```suggestion
| 2 | Integer <br /> BigInt | Float | - | 
```

> | Float | Integer <br /> BigInt | - |

Author: ```suggestion
| 3 | Float | Integer <br /> BigInt | - |
```

> | Int <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |

Author: ```suggestion
| 4 | Integer <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |
```

- For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
- For **other** destinations, the table name is truncated after 127 characters.

### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?

RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.

| Scenario | Incoming data type | Warehouse column data type | Notes |
| :-----| :----| :------| :------|
| 1 | All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
| 2 | Integer <br /> BigInt | Float | - | 
| 3 | Float | Integer <br /> BigInt | - |
| 4 | Integer <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |

<div class="warningBlock">

> The incoming data which <strong>cannot</strong> be typecasted into the warehouse column-specific data type is sent to the <code class="inline-code">rudder_discards</code> table.

Reviewer 2: ```suggestion
In any scenario other than the ones covered above, that is, when the incoming data <strong>cannot</strong> be typecasted into the warehouse column-specific data type, it is sent to the <code class="inline-code">rudder_discards</code> table.
```
